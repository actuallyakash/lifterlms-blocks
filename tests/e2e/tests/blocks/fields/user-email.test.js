/**
 * Test formsReady() events run on domReady()
 *
 * @since 1.12.0
 * @version 1.12.0
 */

import {
	getEditedPostContent,
	insertBlock,
} from '@wordpress/e2e-test-utils';

import {
	logoutUser,
	toggleOpenRegistration,
	visitPage,
} from '@lifterlms/llms-e2e-test-utils';

import {
	clearBlocks,
	visitForm,
} from '../../../util';

describe( 'Blocks/FormFields/UserEmail', () => {

	beforeEach( async () => {

		await visitForm();

	} );

	describe( 'Editor', () => {

		beforeEach( async () => {

			await clearBlocks();
			page.once( 'dialog', async dialog => await dialog.accept() ); // Leave page without saving.

		} );

		it ( 'can be created using the block inserter', async () => {

			await insertBlock( 'User Email' );
			expect( await getEditedPostContent() ).toMatchSnapshot();

		} );

		it ( 'can be created by typing "/user_email"', async () => {

			await page.keyboard.type( '/user_email' );
			await page.waitForXPath(
				`//*[contains(@class, "components-autocomplete__result") and contains(@class, "is-selected") and contains(text(), 'User Email')]`
			);
			await page.keyboard.press( 'Enter' );
			expect( await getEditedPostContent() ).toMatchSnapshot();

		} );

	} );

	describe( 'Frontend', () => {

		it ( 'shows a user email field on the frontend', async () => {

			await toggleOpenRegistration( true );

			await logoutUser();

			await visitPage( 'dashboard' );

			expect( await page.$eval( '#email_address', el => el.parentNode.outerHTML ) ).toMatchSnapshot();

		} );

	} );


} );
