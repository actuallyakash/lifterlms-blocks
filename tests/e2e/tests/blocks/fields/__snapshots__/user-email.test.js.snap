// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks/FormFields/UserEmail Editor can be created by typing "/user_email" 1`] = `"<!-- wp:llms/form-field-user-email {\\"description\\":\\"\\",\\"field\\":\\"email\\",\\"required\\":true,\\"label\\":\\"Email Address\\",\\"label_show_empty\\":false,\\"match\\":\\"email_address_confirm\\",\\"options\\":[],\\"options_preset\\":\\"\\",\\"placeholder\\":\\"\\",\\"name\\":\\"email_address\\",\\"id\\":\\"email_address\\",\\"data_store\\":\\"users\\",\\"data_store_key\\":\\"user_email\\"} /-->"`;

exports[`Blocks/FormFields/UserEmail Editor can be created using the block inserter 1`] = `"<!-- wp:llms/form-field-user-email {\\"description\\":\\"\\",\\"field\\":\\"email\\",\\"required\\":true,\\"label\\":\\"Email Address\\",\\"label_show_empty\\":false,\\"match\\":\\"email_address_confirm\\",\\"options\\":[],\\"options_preset\\":\\"\\",\\"placeholder\\":\\"\\",\\"name\\":\\"email_address\\",\\"id\\":\\"email_address\\",\\"data_store\\":\\"users\\",\\"data_store_key\\":\\"user_email\\"} /-->"`;

exports[`Blocks/FormFields/UserEmail Frontend shows a user email field on the frontend 1`] = `"<div class=\\"llms-form-field type-email llms-cols-12 llms-cols-last llms-is-required\\"><label for=\\"email_address\\">Email Address<span class=\\"llms-required\\">*</span></label><input class=\\"llms-field-input\\" data-match=\\"email_address_confirm\\" id=\\"email_address\\" name=\\"email_address\\" required=\\"required\\" type=\\"email\\"></div>"`;
